applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp

```diff
@@ -597,33 +597,31 @@ void NavierStokesWallCondition<TDim,TNumNodes>::ComputeGaussPointBehrSlipRHSCont
             CompleteNodalSigma[nnode][4] = ShearStressOfElement[4];  // no pressure in shear component
             CompleteNodalSigma[nnode][5] = ShearStressOfElement[5];  // no pressure in shear component
 #ifdef KRATOS_DEBUG
-            if ( abs( ShearStressOfElement[0] ) > 0.001 || abs( ShearStressOfElement[1] ) > 0.001 || abs( ShearStressOfElement[2] ) > 0.001 ){
+            if ( std::abs( ShearStressOfElement[0] ) > 0.001 || std::abs( ShearStressOfElement[1] ) > 0.001 || std::abs( ShearStressOfElement[2] ) > 0.001 ){
                 KRATOS_WARNING("Behr Contribution in SLIP condition") << "The normal components of the viscous stress are still present" << std::endl;
             }
 #endif
         }
     }
 
-    Vector CompleteSigmaInterpolated;
-    CompleteSigmaInterpolated = ZeroVector(3);
+    Vector CompleteSigmaInterpolated = ZeroVector(TNumNodes);
 
-    std::vector<array_1d<double,TNumNodes+1>> NodalEntriesRHS(TNumNodes);
+    std::vector<array_1d<double,TNumNodes>> NodalEntriesRHS(TNumNodes);
 
     // Loop all nodal contributions
     for (unsigned int nnode = 0; nnode < TNumNodes; nnode++){
 
-        NodalEntriesRHS[nnode] = zero_vector<double>(3);
+        NodalEntriesRHS[nnode] = zero_vector<double>(TNumNodes);
```

Comment:
I'd ask you to change that to `= ZeroVector(TNumNodes);`. We are planning to remove our dependency to ublas in the near future and this change will automate that for us once the time comes. Other than that, good work and thanks for the quick response.

I'll test it tomorrow at work and hopefully approve.